### ISSUE TITLE
[Rule Request] Conditional checks itselft

### ISSUE THREAD

**Issue by masters3d** [realm/SwiftLint#2061]
### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.
https://github.com/realm/SwiftLint/issues/1502#issuecomment-300011533

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
if someObj == someObj {} // always true
if objA != objA {} // always false
```
This would not trigger even though they might have the same value. :

```swift
let someObj2 = someObj 
if someObj2 == someObj {} // always true
```

3. Should the rule be configurable, if so what parameters should be configurable?
probably not. 

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.
Probably Default if fast enough.

**Comment by marcelofabri** [comment]
Duplicate of #1371?

**Comment by masters3d** [comment]
ha! totally, I just could not find it.
